river/covariance/emp.py

```diff
@@ -178,35 +194,63 @@ def update_many(self, X: pd.DataFrame):
         mean_arr = X_arr.mean(axis=0)
         cov_arr = np.cov(X_arr.T, ddof=self.ddof)
 
+        n = len(X)
         mean = dict(zip(X.columns, mean_arr))
         cov = {
             (i, j): cov_arr[r, c]
             for (r, i), (c, j) in itertools.combinations_with_replacement(enumerate(X.columns), r=2)
         }
 
-        for i, j in itertools.combinations(sorted(X.columns), r=2):
+        self = self._from_state(n=n, mean=mean, cov=cov, ddof=self.ddof)
+
+        return self
+
+    @classmethod
+    def _from_state(cls, n: int, mean: dict, cov: dict, *, ddof=1):
+        """Create a new instance from state information.
+
+        Parameters
+        ----------
+        cls (type)
+            The class type.
+        n (int)
+            The number of data points.
+        mean (dict)
+            A dictionary of variable means.
+        cov (dict)
+            A dictionary of covariance or variance values.
+        ddof (int, optional)
+            Degrees of freedom for covariance calculation. Defaults to 1.
```

Comment:
There is no need to use type annotations in the doctrings. The docs will take it directly from the method signature :)